logstash-core/lib/logstash/util/dead_letter_queue_factory.rb

```diff
@@ -0,0 +1,32 @@
+require 'logstash/environment'
+
+module LogStash; module Util
+  class DeadLetterQueueWriter
+
+    def initialize(inner_writer)
+      @inner_writer = inner_writer
+    end
+
+    def write(logstash_event, plugin_type, plugin_id, reason)
+      if @inner_writer && @inner_writer.is_open
+        @inner_writer.write_entry(logstash_event.to_java, plugin_type, plugin_id, reason)
+      end
+    end
+  end
+
+  class DeadLetterQueueFactory
+    java_import org.logstash.common.DeadLetterQueueFactory
+
+    def self.get(pipeline_id)
+      if LogStash::SETTINGS.get("dead_letter_queue.enable")
+        return DeadLetterQueueWriter.new(
+            DeadLetterQueueFactory::get_writer(pipeline_id, LogStash::SETTINGS.get("path.dead_letter_queue")))
+      end
+      return DeadLetterQueueWriter.new(nil)
```

Comment:
in other areas we have Null objects for this, like NullMetric.

Can we do a NullDeadLetterQueueWriter here instead of passing nil to a DeadLetterQueueWriter? I wonder if doing so could remove the need for this DeadLetterQueueWriter wrapper class?